src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAll/FixAllState.cs

```diff
@@ -64,29 +83,37 @@ public FixAllState WithCodeActionEquivalenceKey(string? codeActionEquivalenceKey
         public FixAllState WithDocumentAndProject(Document? document, Project project)
             => With(documentAndProject: (document, project));
 
+        public FixAllState WithFixAllSpans(ImmutableArray<TextSpan> fixAllSpans)
+            => With(fixAllSpans: fixAllSpans);
+
         public FixAllState With(
             Optional<(Document? document, Project project)> documentAndProject = default,
             Optional<FixAllScope> scope = default,
-            Optional<string?> codeActionEquivalenceKey = default)
+            Optional<string?> codeActionEquivalenceKey = default,
+            Optional<ImmutableArray<TextSpan>> fixAllSpans = default)
```

Comment:
happy there is a 'With' here :)  i wish that other type was written the same way.